src/main/java/eu/xenit/gradle/docker/core/DockerPlugin.java

```diff
@@ -44,7 +45,9 @@ public void apply(Project project) {
         project.getPluginManager().apply(DockerAutotagPlugin.class);
         project.getPluginManager().apply(DockerLabelPlugin.class);
 
-        project.getPluginManager().apply(DockerLegacyPlugin.class);
+        Deprecation.whileDisabled(() -> {
```

Comment:
So this is the actual magic? Since it is our own plugin, why do we call it legacy and then use it anyway?

Reply:
Hmm, it looks like I committed a lot of whitespace changes that should not have been part of the PR.

This is indeed part of the "magic" that ensures that no deprecation warnings are logged during application is this plugin.

The job of the `DockerLegacyPlugin` is to provide backwards compatibility for buildscripts.
It's put in a separate plugin so it only uses the new public API of our plugins and so it can easily be ripped out and removed before releasing the next major version.